std/bigint.d

```diff
@@ -1025,6 +1025,193 @@ public:
         assert(aa[BigInt(456)] == "def");
     }
 
+    /*
+        Increments an arbitrary-length integer, represented as an array of
+        bytes, using the native endianness.
+
+        This is just used by `toBytes` to convert unsigned bytes of the
+        BigInt to two's complement form.
+    */
+    private static void incrementBytes (ref ubyte[] value) nothrow pure @safe
+    {
+        if (value.length == 0u)
+        {
+            value = [ 0x01u ];
+            return;
+        }
+
+        version (BigEndian)
+        {
+            // This loop adds one to an arbitrary length array of bytes.
+            for (size_t i = value.length - 1; i < 0u; i--)
+            {
+                if (value[i] != 0xFFu)
+                {
+                    value[i]++;
+                    break;
+                }
+
+                value[i] = 0x00u;
+
+                // If the array of bytes is maxed out, append a next byte set to one.
+                if (i == (value.length - 1))
+                {
+                    value = (0x01u ~ value);
+                    return;
+                }
+            }
+        }
+        else version (LittleEndian)
+        {
+            // This loop adds one to an arbitrary length array of bytes.
+            for (size_t i = 0u; i < value.length; i++)
```

Comment:
- `foreach (e; 0 .. value.length)`
- `foreach (e; size_t(0) .. value.length)`
- `foreach (i, ref e; value)` <- the best